src/lib/expansion/expansion-panel.ts

```diff
@@ -0,0 +1,108 @@
+  AfterContentInit,
+  OnDestroy,
+  ContentChildren,
+  QueryList,
+  Optional,
+  forwardRef,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+import {Subscription} from 'rxjs/Subscription';
+import {Subject} from 'rxjs/Subject';
+import 'rxjs/add/operator/takeUntil';
+import {CdkAccordionChild, CdkAccordion} from './accordion';
+
+/** Time and timing curve for expansion panel animations. */
+export const EXPANSION_PANEL_ANIMATION_TIMING = '225ms cubic-bezier(0.4,0.0,0.2,1)';
+
+@Component({
+  moduleId: module.id,
+  styleUrls: ['./expansion-panel.css'],
+  selector: 'md-expansion-panel, mat-expansion-panel',
+  templateUrl: './expansion-panel.html',
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    'class': 'mat-expansion-panel',
+    '[class.mat-expanded]': 'expanded',
+    '[@panelStyle]': 'getPanelStyle()',
+  },
+  providers: [
+    {provide: CdkAccordionChild, useExisting: forwardRef(() => MdExpansionPanel)}
+  ],
+  animations: [
+    trigger('bodyExpansion', [
+      state('collapsed', style({height: '0px', 'padding-bottom': '0'})),
+      state('expanded', style({height: '*', 'padding-bottom': '16px'})),
+      transition('expanded <=> collapsed', animate(EXPANSION_PANEL_ANIMATION_TIMING)),
+    ]),
+    trigger('panelStyle', [
+      state('collapsed', style({margin: '0'})),
+      state('default', style({margin: '16px 0'})),
+      state('flat', style({margin: '0'})),
+      transition('flat <=> collapsed, default <=> collapsed, flat <=> default',
+                 animate(EXPANSION_PANEL_ANIMATION_TIMING)),
+    ]),
+  ],
+})
+export class MdExpansionPanel extends CdkAccordionChild {
+  /** Whether the expansion indicator should be shown. */
+  @Input() showExpandIndicator: boolean = true;
+  /** The expansion panel style. */
+  @Input() panelStyle: string = 'default';
+  /** Whether the expansion panel is expanded. */
+  private _expanded: boolean;
+  @Input()
+  get expanded(): boolean { return this._expanded; }
+  set expanded(expanded: boolean) {
+    // Only emit events and update the internal value if the value changes.
+    if (this._expanded !== expanded) {
+      this._expanded = expanded;
+      if (expanded) {
+        this.open.emit();
+      } else {
+        this.close.emit();
+      }
+    }
+  }
+
+  constructor(@Optional() @Host() public accordion: CdkAccordion) {
+    super();
+  }
+
+  /** Whether the expansion indicator should be shown. */
+  getShowExpandIndicator() {
+    if (this.accordion) {
+      return this.accordion.showExpandIndicator;
+    }
+    return this.showExpandIndicator;
+  }
+
+  /** Gets the panel style. */
+  getPanelStyle(): string {
+    if (!this.expanded) {
+      return this.getExpandedState();
+    }
+    if (this.accordion) {
+      return this.accordion.accordionStyle;
+    }
+    return this.getExpandedState();
+  }
+
+  /** Gets the expanded state string. */
+  getExpandedState(): string {
```

Comment:
Return type should be `'expanded' | 'collapsed'`, probably as a `type` definition.